### PR TITLE
Addition of Project Euler problem 34

### PR DESCRIPTION
**Pull request by mansiag**
**Fixes issue:**
N/A

**Changes:**
Added Project Euler problem 34 in cpp.

### CODE REVIEW

File: code/online_challenges/src/project_euler/problem_034/problem_034.cpp [hunk 1]
```diff
@@ -0,0 +1,27 @@
+#include <iostream>
+#include <vector>
+int factorial (int n)
```
Comment (arnavb) [inline code/online_challenges/src/project_euler/problem_034/problem_034.cpp]: Add newline above.

File: code/online_challenges/src/project_euler/problem_034/problem_034.cpp [hunk 2]
```diff
@@ -0,0 +1,27 @@
+#include <iostream>
+#include <vector>
+int factorial (int n) 
+{
+    if(n == 0)
```
Comment (arnavb) [inline code/online_challenges/src/project_euler/problem_034/problem_034.cpp]: TBH, I would avoid the recursive definition, as recursion in C++ takes up a lot of stack space. I would just use a loop. That would remove the need for the if statement as well. If you want to shorten this and keep this implementation, then you can do the following:

```C++
return (n <= 1) ? (1) : (n * factorial(n - 1));
```

File: code/online_challenges/src/project_euler/problem_034/problem_034.cpp [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+        return 1;
+    return n * factorial(n - 1);
+}
+
+int main()
+{
+    std::vector<int> factor(10);
```
Comment (arnavb) [inline code/online_challenges/src/project_euler/problem_034/problem_034.cpp]: factor is a misleading name; factorials is okay.

File: code/online_challenges/src/project_euler/problem_034/problem_034.cpp [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    for(int i = 0; i < 10; ++i)
+        factor[i] = factorial(i);
+    int num = 3, temp, sum = 0;
+    while (num < 2540162)
```
Comment (arnavb) [inline code/online_challenges/src/project_euler/problem_034/problem_034.cpp]: Where did you get this number from? If it has a significant meaning, assign it to a `constexpr` constant.

File: code/online_challenges/src/project_euler/problem_034/problem_034.cpp [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    {
+        temp = 0;
+        for(size_t i = num; i > 0; i /= 10)
```
Comment (arnavb) [inline code/online_challenges/src/project_euler/problem_034/problem_034.cpp]: `std::size_t`

File: code/online_challenges/src/project_euler/problem_034/problem_034.cpp [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+
+int main()
+{
+    std::vector<int> factor(10);
+    for(int i = 0; i < 10; ++i)
+        factor[i] = factorial(i);
+    int num = 3, temp, sum = 0;
+    while (num < 2540162)
+    {
+        temp = 0;
```
Comment (arnavb) [inline code/online_challenges/src/project_euler/problem_034/problem_034.cpp]: Just move `temp` inside the loop.

File: code/online_challenges/src/project_euler/problem_034/problem_034.cpp [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+        for(size_t i = num; i > 0; i /= 10)
+            temp += factor[i % 10];
+        if(temp == num)
+            sum += num;
+        ++num;
+    }
+    std::cout << sum << "\n";
+    return 0;
```
Comment (arnavb) [inline code/online_challenges/src/project_euler/problem_034/problem_034.cpp]: Remove this.

File: code/online_challenges/src/project_euler/problem_034/problem_034.cpp [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+}
```
Comment (arnavb) [inline code/online_challenges/src/project_euler/problem_034/problem_034.cpp]: Add a newline to the EOF.

**Comment by arnavb** [comment]
Make sure to check the Travis CI Build to see if your file compiled!

**Comment by mansiag** [comment]
@arnavb Is it alright?

**Comment by mansiag** [comment]
Thanks!

**Comment by AdiChat** [comment]
Looks great 😍 

Hope you are enjoying your journey 🚆  with **OpenGenus**

**Pull request merged**